File: Lexor/UI/lexor_desktop/lib/models/legal_document_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'legal_document_response.g.dart';

@JsonSerializable()
class LegalDocumentResponse {
  final int id;
  final String name;
  final String categoryName;
  final DateTime uploadedAt;

  LegalDocumentResponse({
    required this.id,
    required this.name,
    required this.categoryName,
    required this.uploadedAt,
  });

  factory LegalDocumentResponse.fromJson(Map<String, dynamic> json) =>
      _$LegalDocumentResponseFromJson(json);
}

File: Lexor/Lexor.Model/Responses/LegalDocumentResponse.cs
namespace Lexor.Model.Responses
{
    public class LegalDocumentResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string CategoryName { get; set; } = string.Empty;
        public DateTime UploadedAt { get; set; } 

    }
}
